Develop1

TimurSharibov opened the pull request.

TimurSharibov committed:
Созданы тесты для классов

TimurSharibov committed:
Добавлен отчет jacoco

TimurSharibov committed:
Добавлен отчет jacoco

TimurSharibov committed:
Add Readme.md

TimurSharibov commented:
Исправлено.
[nikolay-977](https://github.com/nikolay-977) commented [last week](https://github.com/yandex-praktikum/QA-java-diplom-1/pull/781#issuecomment-2266063017)
Необходимо исправить: при создании пр-а была указана ветка из репозитория заготовки. Это происходит при создании пр-а по умолчанию. При создании пр-а на это необходимо было обратить внимание(об этом упоминается в инструкции). Чтобы это исправить - необходимо в созданном пул-реквесте нажать кнопку edit и указать ветку main своего репозитория вместо ветки main репозитория заготовки.

nikolay-977 commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,19 @@
+    @Test
+    public void testGetPrice() {
+        Bun bun = new Bun("white bun", 200);
+        assertEquals(200, bun.getPrice(), 0.0);
```
Необходимо исправить: не следует использовать литералы, значение которых не является очевидным (антипаттерн магические константы). Лучше сохранить в переменную с наименованием, отражающим суть содержимого.

nikolay-977 commented on src/test/java/DatabaseTest.java:
```diff
@@ -0,0 +1,90 @@
+
+    @Test
+    public void testAvailableBunsContainWhiteBun() {
+        // ���������, ��� ����� ��������� ������� ���� ������� � ������ "white bun"
```
Необходимо исправить кодировку.

TimurSharibov committed:
1. Антипаттерн магические константы удалены и заменены на переменные отражающие суть.
2. Исправлена кодировка на UTF-8.
3. Добавлены комментарии к тестам.
4. Создан и добавлен новый отчет jacoco.
